### ISSUE TITLE
Cannot read empty XML document

### ISSUE THREAD

**Issue by TomWright** [TomWright/dasel#131]
**Describe the bug**
Dasel fails to unmarshal an empty string when using the XML parser.

**To Reproduce**
```bash
echo '' | dasel -p xml .

Error: could not load input: could not unmarshal data: EOF
```

**Expected behavior**
An empty map.
The XML parser should return `nil, nil` when the input is empty.

**Screenshots**

**Desktop (please complete the following information):**
dasel version 1.14.1

**Comment by link4531** [comment]
While I believe this "bug" also includes adding initial values to the xml file.
where the command:  
`dasel put string -p xml '.RootNode' 'Rooty' `

Workaround is to use -p yaml or json followed by re-parsing the file using -w xml.

**Comment by TomWright** [comment]
That's correct. This bug will also appear when providing empty input to any `dasel put` commands.

Another workaround is the following:
```bash
echo '' | dasel put string -r json -w xml '.RootNode' 'Rooty'

<RootNode>Rooty</RootNode>
```

**Comment by TomWright** [comment]
This has been fixed and released under `v1.15.0`.
